plugins/scaffolder-backend/src/scaffolder/actions/builtin/catalog/fetch.test.ts

```diff
@@ -127,4 +111,41 @@ describe('catalog:fetch', () => {
     });
     expect(mockContext.output).not.toHaveBeenCalled();
   });
+
+  it('should return entities from catalog', async () => {
+    getEntitiesByRefs.mockReturnValueOnce({
+      items: [
+        {
+          metadata: {
+            namespace: 'default',
+            name: 'test',
+          },
+          kind: 'Component',
+        } as Entity,
+      ],
+    });
+
+    await action.handler({
+      ...mockContext,
+      input: {
+        entityRefs: ['component:default/test'],
```

Comment:
would like to have the optionality cases tested as well - both where optional = false and optional = true, while fetching missing entities